modules/indexing/src/main/java/org/apache/ignite/internal/processors/query/h2/sql/GridSqlQueryParser.java

```diff
@@ -1323,9 +1340,37 @@ private GridSqlCreateTable parseCreateTable(CreateTable createTbl) {
                 res.affinityKey(pkCols0.iterator().next());
         }
 
+        if (implicitPk) {
+            log.warning("No PRIMARY KEY defined for table '" + data.tableName + "'. " +
+                "An auto-generated implicit primary key will be created.");;
+        }
+
         return res;
     }
 
+    /**
+     * @param createTbl {@code CREATE TABLE} statement.
+     */
+    private void addImplicitPk(CreateTable createTbl) {
```

Comment:
does it make sense to move all this implicit-pk-related staff to the command processor? Right now we have too much places changing theirs behaviour with regard to implicitPk() flag. Why don't leave the parsing and command itself dumb, and change the handling of the command only?

Reply:
Thanks, removed unnecessary column creation from the parser.